src/serializer/ResidualHeapSerializer.js

```diff
@@ -1852,10 +1855,15 @@ export class ResidualHeapSerializer {
   // CreatedObjects -- should take care of itself
   _serializeAdditionalFunction(additionalFunctionValue: FunctionValue, additionalEffects: AdditionalFunctionEffects) {
     let { effects, transforms, generator } = additionalEffects;
-    if (!this.additionalFunctionValueInfos.has(additionalFunctionValue)) {
+    if (
+      !this.additionalFunctionValueInfos.has(additionalFunctionValue) ||
+      // break self-reference cycle
```

Comment:
The serializer hits an invariant otherwise because in the case:

```javascript
function additional() {
   return [additional];
}
```

The emitter emits an empty array and waits for `additional` to be done before emitting the assignment to the array by which point it is in the wrong body.